### PR TITLE
Add experiment api

### PR DESCRIPTION
**Pull request by idonov8**
Will look like that:
Screen Shot 2022-10-23 at 17 23 18

### CODE REVIEW

**Comment by guysmoilov** [comment]
What about the APIs to fetch the list of experiments of a repo, and the metric histories of an experiment?
Without at least the first one, how are you supposed to know the experiment key?

File: spec.yaml [hunk 1]
```diff
@@ -122,7 +122,28 @@ paths:
             application/json:
               schema: 
                 $ref: './components/schemas/Issue.yaml'
-  
+  /experiments/experiment/{experimentKey}/name:
```
Comment (simonlsk) [inline spec.yaml]: What about get experiments?

File: spec.yaml [hunk 2]
```diff
@@ -198,6 +199,176 @@ paths:
...
+        content:
+          application/json:
+            schema:
+              $ref: './components/schemas/ExperimentNamePost.yaml'
+      responses:
+        '200':
+          description: Set name for experiment successfully
+          content:
+            application/json:
+              example: "Awesome New Experiment Name"
```
Comment (guysmoilov) [inline spec.yaml]: This endpoint description is now inaccurate - it's now a general puprpose "edit experiment" API endpoint, that only allows editing the name for now

File: components/schemas/ExperimentNamePost.yaml [hunk 3]
```diff
@@ -0,0 +1,7 @@
+type: object
+properties:
+  name:
+    type: string
+    example: Awesome Experiment
+required:
+  - name
```
Comment (guysmoilov) [inline components/schemas/ExperimentNamePost.yaml]: File should be renamed to reflect that it's a general Edit POST now.

File: components/parameters/ExperimentName.yaml [hunk 4]
```diff
@@ -0,0 +1,6 @@
+name: experimentName
+in: path
+description: A name for the experiment
+required: true
+schema:
+  type: string
```
Comment (guysmoilov) [inline components/parameters/ExperimentName.yaml]: Seems to be unused now

File: spec.yaml [hunk 5]
```diff
@@ -9,6 +9,7 @@ info:
    
 servers:
   - url: https://dagshub.com/api/v1/
+  - url: http://localhost:3000/api/v1/
```
Comment (guysmoilov) [inline spec.yaml]:
```suggestion
```

**Review by guysmoilov — CHANGES_REQUESTED** [review]

**Review by guysmoilov — APPROVED** [review]

**Pull request merged**